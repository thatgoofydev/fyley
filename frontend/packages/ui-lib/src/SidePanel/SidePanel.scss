@import "src/base-styling/vars";

.sidepanel-wrapper {
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  overflow: hidden;
  pointer-events: none;
}

.mask {
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
  background-color: $sidepanel-mask-bg;
  pointer-events: auto;

  transition: background-color .5s;
}

.sidepanel {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  width: $sidepanel-width;
  background-color: $sidepanel-bg;
  pointer-events: auto;

  margin-right: 0;
  transition: margin-right .5s;
}

.closed {
  .mask {
    //display: none;
    pointer-events: none;
    background-color: transparent;
  }

  .sidepanel {
    margin-right: -$sidepanel-width;
  }
}