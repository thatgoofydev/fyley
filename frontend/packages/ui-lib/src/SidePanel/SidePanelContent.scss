@import "src/base-styling/vars";

.sidepanel-content {
  padding: $sidepanel-content-padding;
}