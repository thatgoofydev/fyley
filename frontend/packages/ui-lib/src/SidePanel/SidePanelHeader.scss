@import "src/base-styling/vars";

.sidepanel-header {
  padding: $sidepanel-header-padding;
  border-bottom: $sidepanel-header-border;
  display: flex;
  justify-content: space-between;

  h2 {
    margin: 0;
  }

  .close:after {
    content: "\00d7";
    font-size: 2em;
    line-height: 28px;
    cursor: pointer;
  }
}