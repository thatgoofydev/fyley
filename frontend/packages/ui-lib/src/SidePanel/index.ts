export { SidePanel } from "./SidePanel";
export { SidePanelHeader } from "./SidePanelHeader";
export { SidePanelContent } from "./SidePanelContent";
