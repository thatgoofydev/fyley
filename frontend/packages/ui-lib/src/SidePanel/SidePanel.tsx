import React, { FC } from "react";
import "./SidePanel.scss";
import classNames from "classnames";

export interface IProps {
  isOpen: boolean;
  onClose?: () => void;
  "data-testid"?: string;
}

export const SidePanel: FC<IProps> = ({ isOpen, onClose, children, "data-testid": testId }) => {
  const handleMaskClick = () => {
    if (onClose) onClose();
  };

  return (
    <>
      <div className={classNames("sidepanel-wrapper", { closed: !isOpen })} data-testid={testId}>
        <div className="mask" onClick={handleMaskClick} />
        <div className="sidepanel">{children}</div>
      </div>
    </>
  );
};
