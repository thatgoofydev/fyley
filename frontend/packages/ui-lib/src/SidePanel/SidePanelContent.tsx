import React, { FC } from "react";
import "./SidePanelContent.scss";

export const SidePanelContent: FC = ({ children }) => {
  return <div className="sidepanel-content">{children}</div>;
};
