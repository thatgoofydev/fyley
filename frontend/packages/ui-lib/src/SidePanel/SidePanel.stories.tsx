import React, { useState } from "react";
import { Meta, Story } from "@storybook/react";
import { SidePanel } from "./SidePanel";
import { Sheet, SheetContent } from "../Sheet";
import { SidePanelHeader } from "./SidePanelHeader";
import { SidePanelContent } from "./SidePanelContent";

export default {
  title: "Pages/SidePanel"
} as Meta;

export const SidePanelExample: Story = _ => {
  const [isOpen, setOpen] = useState(false);

  const openSidePanel = () => setOpen(true);
  const closeSidePanel = () => setOpen(false);

  return (
    <Sheet>
      <SheetContent>
        <button onClick={openSidePanel}>Open SidePanel</button>

        <SidePanel isOpen={isOpen} onClose={closeSidePanel}>
          <SidePanelHeader title="Some title" onClose={closeSidePanel} />
          <SidePanelContent>Some sidebar content</SidePanelContent>
        </SidePanel>
      </SheetContent>
    </Sheet>
  );
};
