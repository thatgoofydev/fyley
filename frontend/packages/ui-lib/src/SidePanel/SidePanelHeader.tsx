import React, { FC } from "react";
import "./SidePanelHeader.scss";

interface IProps {
  title: string;
  onClose?: () => void;
}

export const SidePanelHeader: FC<IProps> = ({ title, onClose }) => {
  const handleCloseClicked = () => {
    if (onClose) onClose();
  };

  return (
    <div className="sidepanel-header">
      <div className="title">
        <h2>{title}</h2>
      </div>
      <div className="close" onClick={handleCloseClicked} />
    </div>
  );
};
