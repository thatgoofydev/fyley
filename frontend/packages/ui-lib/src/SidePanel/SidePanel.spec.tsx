import React from "react";
import { render, cleanup } from "@testing-library/react";
import { SidePanel } from "./SidePanel";
import { SidePanelHeader } from "./SidePanelHeader";
import { SidePanelContent } from "./SidePanelContent";

afterEach(cleanup);

describe("SidePanel", function () {
  it("should render closed if !isOpen", () => {
    const { container, getByTestId } = render(
      <SidePanel isOpen={false} data-testid="panel">
        <SidePanelHeader title="some title" />
        <SidePanelContent>
          <p>Some content inside of the panel...</p>
        </SidePanelContent>
      </SidePanel>
    );

    expect(container).toMatchSnapshot();
    expect(getByTestId("panel").classList).toContain("closed");
  });

  it("should render open if isOpen", () => {
    const { container, getByTestId } = render(
      <SidePanel isOpen={true} data-testid="panel">
        <SidePanelHeader title="some title" />
        <SidePanelContent>
          <p>Some content inside of the panel...</p>
        </SidePanelContent>
      </SidePanel>
    );

    expect(container).toMatchSnapshot();
    expect(getByTestId("panel").classList).not.toContain("closed");
  });
});
