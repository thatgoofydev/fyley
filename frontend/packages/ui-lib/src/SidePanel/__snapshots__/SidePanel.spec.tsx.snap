// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidePanel should render closed if !isOpen 1`] = `
<div>
  <div
    class="sidepanel-wrapper closed"
    data-testid="panel"
  >
    <div
      class="mask"
    />
    <div
      class="sidepanel"
    >
      <div
        class="sidepanel-header"
      >
        <div
          class="title"
        >
          <h2>
            some title
          </h2>
        </div>
        <div
          class="close"
        />
      </div>
      <div
        class="sidepanel-content"
      >
        <p>
          Some content inside of the panel...
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`SidePanel should render open if isOpen 1`] = `
<div>
  <div
    class="sidepanel-wrapper"
    data-testid="panel"
  >
    <div
      class="mask"
    />
    <div
      class="sidepanel"
    >
      <div
        class="sidepanel-header"
      >
        <div
          class="title"
        >
          <h2>
            some title
          </h2>
        </div>
        <div
          class="close"
        />
      </div>
      <div
        class="sidepanel-content"
      >
        <p>
          Some content inside of the panel...
        </p>
      </div>
    </div>
  </div>
</div>
`;
